src/core_plugins/markdown_vis/public/markdown_vis_controller.js

```diff
@@ -1,18 +1,28 @@
 import MarkdownIt from 'markdown-it';
-import { uiModules } from 'ui/modules';
-import 'angular-sanitize';
 
 const markdownIt = new MarkdownIt({
   html: false,
   linkify: true
 });
 
-const module = uiModules.get('kibana/markdown_vis', ['kibana', 'ngSanitize']);
-module.controller('KbnMarkdownVisController', function ($scope) {
-  $scope.$watch('vis.params.markdown', function (html) {
-    if (html) {
-      $scope.html = markdownIt.render(html);
+export class MarkdownVisController {
+  constructor(node, vis) {
+    this._vis = vis;
+    this._containerNode = document.createElement('div');
+    this._containerNode.className = 'markdown-vis';
+    node.appendChild(this._containerNode);
+  }
+
+  async render(data, status) {
+    if (status.params && this._vis.params.markdown) {
+      const markdownElement = document.createElement('div');
+      markdownElement.className = 'markdown-body';
+      markdownElement.style['font-size'] = `${this._vis.params.fontSize}pt`;
+      markdownElement.dataset.testSubj = 'markdownBody';
+      markdownElement.innerHTML = markdownIt.render(this._vis.params.markdown);
+      this._containerNode.innerHTML = '';
```

Comment:
Could we use the [more performant way](https://stackoverflow.com/questions/3955229/remove-all-child-elements-of-a-dom-node-in-javascript) instead of setting `innerHTML` to empty string:

```js
while (this._containerNode.firstChild) {
  this._containerNode.removeChild(this._containerNode.firstChild);
}
```

Reply:
actually, it seems that innerHTML is faster (latest chrome and firefox)
https://jsperf.com/innerhtml-vs-removechild/133